src/docs/sdk/research/performance/index.mdx

```diff
@@ -192,6 +192,160 @@ In the example above, if an error bubbles up the call stack we want to be able t
+
+Sentry's model, unlike OpenTelemetry's, makes a hard distinction between two types of span: transaction spans (often refered to as "transactions") and regular spans.
+
+In memory, transaction spans and regular spans have one distinction: transaction spans have one extra attribute, the transaction `name`.
+
+When serialized as JSON for ingestion, though, the differences are greater. Sentry SDKs must serialize regular spans to JSON in a format that directly resembles the in-memory spans. However, the serialization of a transaction span requires mapping its span attributes to a Sentry `Event` (originally used to report errors, expanded to fit new fields of exclusive use for transactions), while all child spans go as a list embedded in the `Event`.
+
+### Transaction Spans Gain Event Attributes
+
+When transactions are transformed from their in-memory representation to an `Event`, they gain more attributes that are not assignable to regular spans, such as `breadcrumbs`, `extra`, `contexts`, `event_id`, `fingerprint`, `release`, `environment`, `user`, etc.
+
+### Lifecycle Hooks
+
+Sentry SDKs expose a `BeforeSend` hook for error events, which allows users to modify and/or drop events before they are sent to Sentry.
+
+When the new type of event was introduced, `transaction`, it was soon decided that they would not go through the `BeforeSend` hook, for essentially two reasons:
+
+- Prevent user code from relying on the dual form of transactions (sometimes looking like a span, sometimes like an event, see earlier sections above);
+- Avoid existing `BeforeSend` functions that were written with only errors in mind from interfering with transactions, be it mutating them accidentally, dropping them altogether, or some other unexpected side-effect.
+
+However, it was also clear that some form of lifecycle hook was necessary to allow users to do things like updating a transaction name.
+
+We ended up with the middle ground of allowing the mutation/dropping of transaction events through `EventProcessor` (a more general form of `BeforeSend`), which solves problems by giving users immediate access to their data before it goes out of the SDK, but is also the worse of both worlds in that it exposes the transaction duality that was never intended to leak and is more complicated to use than `BeforeSend`.
+
+In OpenTelemetry, spans go through span processors, which are two lifecycle hooks: one when a span is started and one when it is ended.
+
+### Nested Transactions
+
+On one hand, Sentry's ingestion model does not allow for nested transactions. Transactions are meant to mark service transitions.
+
+On the other hand, Sentry as a product requires that transactions exist. Data outside of a transaction is lost.
+
+In practice, SDKs have no way of preventing transactions from becoming nested. The end result is possibly surprising to users, as each transaction starts a new tree. The only way to relate those trees is through the `trace_id`.
```

Comment:
> 1. When you're talking about spans being "nested," it seems like what you really mean is "simultaneous." To my mind, they're not nested unless they actually _are_ correctly associated with each other by using the same trace id.

Spans were "made to be nested". Transactions were not (made to be nested in the same service).

By "nested" I mean the two transactions have an ancestor-descendant relationship and are produced in the same service. Note that in the one-transaction-per-service model, as in the example in the beginning of the section, transactions are obvious nested within a trace. 


> 2\. I'm not sure how much this matters, but IIRC SDKs have different behavior here. For example, in JS if you have transaction A on the scope and you replace it with transaction B, transaction A is lost unless there's something else referencing it, because when B ends it just stays there, and even if it does get popped off, A never gets put back on. In Python, OTOH, context managers do pop off B, and they then put A back.

^^ this shows how, in different ways, "nested transactions" are not well-supported. Still, there is nothing stopping anyone from calling `startTransaction -> startChild -> startTransaction (oops) -> ...`